src/releases/semver/mod.rs

```diff
@@ -0,0 +1,601 @@
+    pub(crate) package: Package,
+}
+
+/// Bump the version of a single `package` using `rule`.
+pub(super) fn bump_version(
+    rule: &Rule,
+    dry_run: bool,
+    package: Package,
+) -> Result<PackageVersion, StepError> {
+    let versions = get_version(&package)?;
+    let version = bump(versions, rule)?;
+    let package = set_version(package, &version, dry_run)?;
+    Ok(PackageVersion { version, package })
+}
+
+/// The implementation of [`crate::step::Step::BumpVersion`].
+///
+/// Bumps the version of every configured package using `rule`.
+pub(crate) fn bump_version_and_update_state(
+    run_type: RunType,
+    rule: &Rule,
+) -> Result<RunType, StepError> {
+    let (mut dry_run_stdout, mut state) = match run_type {
+        RunType::DryRun { state, stdout } => (Some(stdout), state),
+        RunType::Real(state) => (None, state),
+    };
+
+    for package in state.packages.iter().cloned() {
+        let PackageVersion { package, version } =
+            bump_version(rule, dry_run_stdout.is_some(), package)?;
+        if let Some(stdout) = dry_run_stdout.as_mut() {
+            writeln!(
+                stdout,
+                "Would bump {name} to version {version}",
+                name = package.name.as_deref().unwrap_or("package"),
+                version = version
+            )?;
+        }
+        state.releases.push(state::Release::Bumped {
+            version,
+            package_name: package.name.clone(),
+        });
+    }
+    if let Some(stdout) = dry_run_stdout {
+        Ok(RunType::DryRun { state, stdout })
+    } else {
+        Ok(RunType::Real(state))
+    }
+}
+
+/// Get the current version of a package.
+pub(crate) fn get_version(package: &Package) -> Result<CurrentVersions, StepError> {
+    let version_from_files = package
+        .versioned_files
+        .iter()
+        .map(|versioned_file| versioned_file.get_version(package.name.as_deref()))
+        .map(|result| result.and_then(|version_string| Version::from_str(&version_string)))
+        .reduce(|accumulator, version| match (version, accumulator) {
+            (Ok(version), Ok(accumulator)) => {
+                if version == accumulator {
+                    Ok(accumulator)
+                } else {
+                    Err(StepError::InconsistentVersions(
+                        version.to_string(),
+                        accumulator.to_string(),
+                    ))
+                }
+            }
+            (_, Err(err)) | (Err(err), _) => Err(err),
+        })
+        .transpose()?;
+
+    let mut current_versions = get_current_versions_from_tag(package.name.as_deref())?;
+
+    if let Some(version_from_files) = version_from_files {
+        if version_from_files.pre.is_none() {
+            current_versions.replace_stable_if_newer(version_from_files);
+        } else {
+            current_versions.insert_prerelease(version_from_files);
+        }
```

Comment:
Why not have a single method that takes a `Version` and updates `CurrentVersions` according to whether the `Version` is a prerelease? (Basically, what if this was moved into a method?)